compiler/damlc/lib/DA/Cli/Damlc.hs

```diff
@@ -603,68 +610,156 @@ execInspectDar inFile = do
+                              (fakeDynFlags
+                                   {pkgState = pdfPkgState packageState})
+                              iuid
+                        ]
+                    pure (src, iuid, depends pkgInfo)
+        let unitIdsTopoSorted = reverse $ topSort depGraph
+        createDirectoryIfMissing True genDir
+        projectPkgDb <- makeAbsolute projectPackageDatabase
+        forM_ unitIdsTopoSorted $ \vertex -> withCurrentDirectory genDir $ do
+            let (src, iuid, _) = vertexToNode vertex
+            unless
+                -- TODO (drsk) remove this filter
+                (iuid `elem`
+                 map stringToInstalledUnitId ["daml-prim", "daml-stdlib"]) $ do
+                  -- typecheck and generate interface files
+                  forM_ src $ \(fp, content) -> do
+                      let path = fromNormalizedFilePath fp
+                      createDirectoryIfMissing True $ takeDirectory path
+                      writeFile path content
+                  opts' <-
+                      mkOptions $
+                      opts
+                          { optWriteInterface = True
+                          , optPackageDbs = [projectPkgDb]
+                          , optIfaceDir =
+                                Just
+                                    (dbPath </> installedUnitIdString iuid)
+                          , optIsGenerated = True
+                          , optMbPackageName =
+                                Just $ installedUnitIdString iuid
+                          }
+                  withDamlIdeState opts' loggerH diagnosticsLogger $ \ide ->
+                      forM_ src $ \(fp, _content) -> do
+                          mbCore <-
+                              runAction ide $
+                              getGhcCore fp
+                          case mbCore of
+                              Nothing ->
+                                  ioError $
+                                  userError $
+                                  "Compilation of generated source for " <>
+                                  installedUnitIdString iuid <>
+                                  " failed."
+                              Just _core -> pure ()
+        -- get the package name and the lf-package
+        [(pkgName1, pkgId1, lfPkg1), (pkgName2, pkgId2, lfPkg2)] <-
+            forM [inFile1, inFile2] $ \inFile -> do
+                bytes <- B.readFile inFile
+                let pkgName = takeBaseName inFile
+                let dar = toArchive $ BSL.fromStrict bytes
+              -- get the main pkg
+                manifest <- getEntry "META-INF/MANIFEST.MF" dar
+                mainDalfPath <-
+                    maybe
+                        (ioError $
+                         userError
+                             "Missing Main-Dalf entry in META-INF/MANIFEST.MF")
+                        pure $
+                    headMay
```

Comment:
`headNote` or `headDef` might be more appropriate here.

Reply:
changed to headNote.